src/gengowatcher/web.py

```
@@ -858,6 +1124,69 @@ async def execute_command(
+    """Download a file from the local file transfer directory."""
+    if not api_instance:
+        raise HTTPException(status_code=503, detail="API not initialized")
+    if not api_instance._is_valid_stored_name(stored_name):
+        raise HTTPException(status_code=400, detail="Invalid file name")
+    entry = api_instance.get_file_entry(stored_name)
+    if entry is None:
+        raise HTTPException(status_code=404, detail="File not found")
+    try:
+        path = api_instance._ensure_within_storage_dir(Path(entry.path))
+    except ValueError:
+        raise HTTPException(status_code=404, detail="File not found")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)